### PR TITLE
Use fewer processes while building OpenBLAS

### LINKED ISSUE #15716: Building Julia uses too many processes
While installing Julia 0.4.5 on Stampede at TACC, I received this message from the system administrators:

```
You are still running too many tasks in the make, please 
STOP performing so many compile tasks at one time, you
are causing excessive load on the logins.  Thanks, TACC Admins
eschnett 18646 18633  0 12:07 pts/366  00:00:00 make -j4
eschnett 33142  9819  1 12:16 pts/366  00:00:00 make -j 16 GOTOBLAS_MAKEFILE= -C kernel TARGET_CORE=NEHALEM kernel
```

It seems that OpenBLAS builds with 16 processes (the number of cores on the login node), although I told Julia to use at most 4 processes (see the first `make` line quoted above).

Using many processes is fine when building on a laptop or private workstation, but on shared resources, one needs to be less possessive.

It seems that these lines in OpenBLAS's `getarch.c` are responsible for this:
```C
#if NO_PARALLEL_MAKE==1
    printf("MAKE += -j 1\n");
#else
#ifndef OS_WINDOWS
    printf("MAKE += -j %d\n", get_num_cores());
#endif
```

### PR DESCRIPTION
**Pull request by eschnett**
Closes #15716

### CODE REVIEW

File: deps/Makefile [hunk 1]
```diff
@@ -1185,6 +1185,7 @@ OPENBLAS_BUILD_OPTS += NO_AVX2=1
 endif
 
 $(BUILDDIR)/$(OPENBLAS_SRC_DIR)/config.status: $(BUILDDIR)/$(OPENBLAS_SRC_DIR)/Makefile
+	cd $(dir $@) && patch -p1 < ../openblas-make.patch
 ifeq ($(OS),WINNT)
 	cd $(dir $@) && patch -p1 < $(SRCDIR)/openblas-win64.patch
```
Comment (tkelman) [inline deps/Makefile]: you'll have to delete this patch, and update the checksums if we're going to update to 0.2.17

File: deps/Makefile [hunk 2]
```diff
@@ -1185,6 +1185,7 @@ OPENBLAS_BUILD_OPTS += NO_AVX2=1
 endif
 
 $(BUILDDIR)/$(OPENBLAS_SRC_DIR)/config.status: $(BUILDDIR)/$(OPENBLAS_SRC_DIR)/Makefile
+	cd $(dir $@) && patch -p1 < ../openblas-make.patch
```
Comment (tkelman) [inline deps/Makefile]: `$(SRCDIR)/openblas-make.patch`

**Comment by tkelman** [comment]
`make -C deps NO_GIT=1 configure-openblas && git rm -f deps/checksums/openblas-53e849f4fcae4363a64576de00e982722c7304f9.tar.gz && git add deps/checksums/openblas-a71e8c82f6a9f73093b631e5deab1e8da716b61f`

**Comment by eschnett** [comment]
Thanks for the pointer to these other checksums.

**Comment by tkelman** [comment]
Note that we should probably merge this in very close succession with #15724 if we want to do both of them, since either in isolation will cause an openblas rebuild.

**Comment by tkelman** [comment]
upstream merged patch in https://github.com/xianyi/OpenBLAS/pull/829 was a little different, uses a different flag name

**Comment by eschnett** [comment]
@tkelman We can apply the patch nonetheless; once a new version of OpenBLAS is released, we will remove our patch anyway.

**Comment by tkelman** [comment]
But the upstream patch uses a different name for the make flag. So we'd need to change. Let's use a patch that matches what we will get from upstream so the flag can match as well.

**Comment by eschnett** [comment]
Now using the upstream commit as patch.

**Comment by tkelman** [comment]
Cool. I'm okay with this version, but for the sake of not making everyone rebuild openblas twice let's see if there are any objections to https://github.com/JuliaLang/julia/pull/15724.

**Pull request merged**